### DISCUSSION TITLE
Safe video app

### DISCUSSION THREAD

**Discussion by farnoodma** [WeAreMahsaAmini/Core#8] [Ideas]
**Context**: According to historic events, the regime always uses videos and pictures posted by people to find the leaders and those who were involved in illegal activities such as protesting. They also fetch those files' metadata to find the person who distributed them. Based on these data, some protestors get arrested and imprisoned.

**State**: Right now some videos are covering people's faces to ensure the privacy of those who were involved, however, still most of the videos show faces without any cover.

**Idea**: Create an app to delete the metadata of a file + cover faces with the help of face recognition algorithms. People can use this to first secure their videos and then send them to social media.

**Comment by ghost** [comment]
[Here](https://github.com/deepinsight/insightface) are some pre-trained deep learning models for face detection. If you want to make an app, getting these models working inside Android is a huge pain, and will probably not work on older phones. If you want to create a website that does this, you need a GPU server which is pricey.

By the way, aren't most people wearing masks during protests?

**Comment by amastaneh** [comment]
Also, There might have been a misunderstanding, but sometimes videos are intended to show the oppressors' faces. 
How we may cover the oppressed while exposing the abuser?

**Reply by ghost** [reply]
They can be detected based on their uniform, but that's a big project, because we would need to gather and annotate training data.

**Comment by freedomfighterforiran** [comment]
I think [PutMask](https://play.google.com/store/apps/details?id=com.putmask.facefilter&hl=fr&gl=US) can be a solution? It is not perfect, but it is better than nothing. You can find more information about it in the following:
1- [Official websiter](https://putmask.com/)  -> They have a Farsi version as well
2- Some [tutorials](https://putmask.com/tutorials) on how to use it -> Again it is provided in different languages

Someone is needed to check whether the application itself is secure or not. At a glance, from the information given in the section [Privacy Policy](https://putmask.com/privacy-policy) appears to be safe.
